velox/expression/VectorUdfTypeSystem.h

```diff
@@ -575,6 +640,40 @@ struct VectorReader<Row<T...>> {
     return !decoded_.isNullAt(offset);
   }
 
+  inline bool containsNull(vector_size_t index) const {
+    if (decoded_.isNullAt(index)) {
+      return true;
+    }
+
+    bool fieldsContainNull = false;
+    std::apply(
+        [&](const auto&... reader) {
+          fieldsContainNull |= (reader->containsNull(index) || ...);
+        },
+        childReaders_);
+
+    return fieldsContainNull;
+  }
+
+  inline bool containsNull(vector_size_t startIndex, vector_size_t endIndex)
+      const {
+    for (auto index = startIndex; index < endIndex; ++index) {
+      if (decoded_.isNullAt(index)) {
+        return true;
+      }
+    }
+
+    bool fieldsContainNull = false;
+    std::apply(
+        [&](const auto&... reader) {
+          fieldsContainNull |=
+              (reader->containsNull(startIndex, endIndex) || ...);
```

Comment:
this is tricky since we want to call containsNull on the reader for each row at he decoded index. 
this one wont do it

Reply:
so the decoded index is the same for all children at a specific row but differ per row.